wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.user.server;
+
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.user.server.dao.PreferenceDao;
+import org.eclipse.che.api.user.server.dao.Profile;
+import org.eclipse.che.api.user.server.dao.User;
+import org.eclipse.che.api.user.server.dao.UserDao;
+import org.eclipse.che.api.user.server.dao.UserProfileDao;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static org.eclipse.che.api.user.server.Constants.ID_LENGTH;
+import static org.eclipse.che.api.user.server.Constants.PASSWORD_LENGTH;
+import static org.eclipse.che.commons.lang.NameGenerator.generate;
+
+/**
+ * Facade for User related operations.
+ *
+ * @author Max Shaposhnik (mshaposhnik@codenvy.com)
+ */
+public class UserManager {
+
+    private final UserDao        userDao;
+    private final UserProfileDao profileDao;
+    private final PreferenceDao  preferenceDao;
+
+
+    @Inject
+    public UserManager(UserDao userDao,
+                       UserProfileDao profileDao,
+                       PreferenceDao preferenceDao) {
+        this.userDao = userDao;
+        this.profileDao = profileDao;
+        this.preferenceDao = preferenceDao;
+    }
+
+
+    /**
+     * Creates new user.
+     *
+     * @param user
+     *         - POJO representation of user entity
+     * @throws ConflictException
+     *         when given user cannot be created
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public void create(User user, boolean isTemporary) throws ConflictException, ServerException, BadRequestException, NotFoundException {
+        user.withId(generate("user", ID_LENGTH))
+            .withPassword(firstNonNull(user.getPassword(), generate("", PASSWORD_LENGTH)));
+        checkPassword(user.getPassword());
+        userDao.create(user);
+
+        profileDao.create(new Profile(user.getId()));
```

Comment:
What will happen if 
userDao.create(user); runs successfully
profileDao.create(new Profile(user.getId())); fails
?
Will user be in inconsistent state or in normal state on get user or get user profile?